patchwork-registries/src/main/java/net/minecraftforge/registries/ForgeRegistry.java

```diff
@@ -70,17 +137,21 @@ public void register(V value) {
 
 		Optional<V> potentialOldValue = vanilla.getOrEmpty(identifier);
 
-		potentialOldValue.ifPresent(
-				oldValue -> {
-					if (oldValue == value) {
-						LOGGER.warn(REGISTRIES, "Registry {}: The object {} has been registered twice for the same name {}.", this.superType.getSimpleName(), value, identifier);
-
-						return;
-					} else {
-						throw new IllegalArgumentException(String.format("The name %s has been registered twice, for %s and %s.", identifier, oldValue, value));
-					}
-				}
-		);
+		if (potentialOldValue.isPresent()) {
+			V oldValue = potentialOldValue.get();
```

Comment:
I need to do multiple checks here, I feel it is easier to control the flow when ifPresent() is removed here.

Reply:
At this point it might make more sense to just do `this.oldValue = vanilla.getOrEmpty(identifier).orElse(null);` and then work with the nullable value directly, since that's effectively what's going on here. It would probably simplify the control flow a bit.

Reply:
solved